torchcde/misc.py

```diff
@@ -119,3 +119,33 @@ def validate_input_path(x, t):
                          "time dimension of size {}.".format(tuple(t.shape), t.size(0)))
 
     return t
+
+
+def forward_fill(data):
+    """Forward fills data in a torch tensor of shape [N, L, C] along the L dim.
+
+    Args:
+        data (torch.Tensor):
+
+    Returns:
+        tensor: Tensor with forward filled data.
+    """
+    # Checks
+    assert isinstance(data, torch.Tensor)
+    assert data.dim() >= 2
+
+    if torch.isnan(data).any():
+        # Reshape to [-1, L]
+        L, C = data.size()[-2:]
+        data_shaped = data.transpose(-1, -2).reshape(-1, L)
+        # Fill the 2d tensor along the length dim
+        mask = torch.isnan(data_shaped)
+        idx = torch.where(~mask, torch.arange(mask.shape[1]), 0)
```

Comment:
I don't believe you've actually tested this. This will throw an error because 0 is not a Tensor.
The ~ is unnecessary because you can flip True/False cases.

Does the `arange` not need to have the device specified?

Stylisticly: prefer $tensor.size(x)$ over $tensor.shape[x]$.